Nil pointer crash pressing a during initial load with dirty submodule (>2000 commits)

### Describe the bug
Opening lazygit in a repo with a dirty submodule containing 2000+ commits, then pressing a (stage all) before initial loading finishes, crashes with a nil pointer dereference.

### To Reproduce
Steps to reproduce the behavior:

1. Have aepo with a submodule having 2000+ commits, and changes made inside the submodule (main repo shows  M sub)
2. Run lazygit in the main repo
3. Immediately press `a` before loading completes
4. panic: runtime error: invalid memory address or nil pointer dereference

### Expected behavior
Keypress is ignored/deferred until the file tree is ready, or handled safely.

### Screenshots
If applicable, add screenshots to help explain your problem.

### Version info:
* commit=v0.64.1, build date=2026-08-12T20:54:55Z, build source=binaryRelease, version=0.64.1, os=linux, arch=amd64, git version=2.55.0

### Terminal info:
Ghostty 1.3.1-arch2

Version
  - version: 1.3.1-arch2
  - channel: tip
Build Config
  - Zig version   : 0.15.2
  - build mode    : .ReleaseFast
  - app runtime   : .gtk
  - font engine   : .fontconfig_freetype
  - renderer      : renderer.generic.Renderer(renderer.OpenGL)
  - libxev        : io_uring
  - kernel version: 7.1.8-arch1-3
  - desktop env   : other
  - GTK version   :
    build         : 4.22.2
    runtime       : 4.22.4
  - libadwaita    : enabled
    build         : 1.9.0
    runtime       : 1.9.3
  - libX11        : enabled
  - libwayland    : enabled
### Additional context
Add any other context about the problem here.

> [!NOTE]
> Please try updating to the latest version or [manually building](https://github.com/jesseduffield/lazygit/#manual) the latest `master` to see if the issue still occurs.

* Yes also occurs on master with commit af04698d9a8bc842fe5b7846492d341faf52f213

<!--
If you want to try and debug this issue yourself, you can run `lazygit --debug` in one terminal panel and `lazygit --logs` in another to view the logs.
-->

```backtrace from the --debug terminal
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xbfe418]

goroutine 1 [running]:
github.com/jesseduffield/lazygit/pkg/gui/filetree.(*FileNode).GetHasInlineMergeConflicts(0x0)
        /home/hrzlgnm/code/lazygit/pkg/gui/filetree/file_node.go:50 +0x18
github.com/jesseduffield/lazygit/pkg/gui/controllers.(*FilesController).toggleStaged(0x1ccfb8643880, {0x1ccfb8712190, 0x1, 0x1}, {0x1211632, 0xf}, {0x1215b43, 0x11}, 0x1ccfb84e6130, 0x1ccfb84e61b8)
        /home/hrzlgnm/code/lazygit/pkg/gui/controllers/files_controller.go:518 +0xda
github.com/jesseduffield/lazygit/pkg/gui/controllers.(*FilesController).toggleStagedAllWithLock(0x1ccfb8643880)
        /home/hrzlgnm/code/lazygit/pkg/gui/controllers/files_controller.go:957 +0x21e
github.com/jesseduffield/lazygit/pkg/gui/controllers.(*FilesController).toggleStagedAll(0x1ccfb8643880)
        /home/hrzlgnm/code/lazygit/pkg/gui/controllers/files_controller.go:920 +0x35
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).callKeybindingHandler(0x1ccfb83b3608, 0x1ccfb82e6790)
        /home/hrzlgnm/code/lazygit/pkg/gui/keybindings.go:416 +0x1ff
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).SetKeybinding.func1(0x1ccfb8324fc8, 0x1ccfb844c908)
        /home/hrzlgnm/code/lazygit/pkg/gui/keybindings.go:386 +0x3f
github.com/jesseduffield/lazygit/pkg/gocui.(*Gui).execKeybinding(0x1ccfb8324fc8, 0x1ccfb844c908, 0x1ccfb8146680)
        /home/hrzlgnm/code/lazygit/pkg/gocui/gui.go:2006 +0x3d
github.com/jesseduffield/lazygit/pkg/gocui.(*Gui).execKeybindings(0x1ccfb8324fc8, 0x1ccfb844c908, 0x1ccfb84e6cc8)
        /home/hrzlgnm/code/lazygit/pkg/gocui/gui.go:1969 +0x4b9
github.com/jesseduffield/lazygit/pkg/gocui.(*Gui).onKey(0x1ccfb8324fc8, 0x1ccfb84e6cc8)
        /home/hrzlgnm/code/lazygit/pkg/gocui/gui.go:1699 +0x1b0
github.com/jesseduffield/lazygit/pkg/gocui.(*Gui).handleEvent(0x1ccfb8324fc8, 0x1ccfb84e6cc8)
        /home/hrzlgnm/code/lazygit/pkg/gocui/gui.go:1173 +0x2a5
github.com/jesseduffield/lazygit/pkg/gocui.(*Gui).processEvent(0x1ccfb8324fc8)
        /home/hrzlgnm/code/lazygit/pkg/gocui/gui.go:1092 +0x51c
github.com/jesseduffield/lazygit/pkg/gocui.(*Gui).MainLoop(0x1ccfb8324fc8)
        /home/hrzlgnm/code/lazygit/pkg/gocui/gui.go:1057 +0x1ba
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).Run(0x1ccfb83b3608, {{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}})
        /home/hrzlgnm/code/lazygit/pkg/gui/gui.go:995 +0x845
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).RunAndHandleError.func1()
        /home/hrzlgnm/code/lazygit/pkg/gui/gui.go:1006 +0x69
github.com/jesseduffield/lazygit/pkg/utils.SafeWithError(0x1ccfb83a5438)
        /home/hrzlgnm/code/lazygit/pkg/utils/utils.go:81 +0x69
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).RunAndHandleError(0x1ccfb83b3608, {{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}})
        /home/hrzlgnm/code/lazygit/pkg/gui/gui.go:1005 +0x132
github.com/jesseduffield/lazygit/pkg/app.(*App).Run(0x1ccfb839cd40, {{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}})
        /home/hrzlgnm/code/lazygit/pkg/app/app.go:285 +0xb8
github.com/jesseduffield/lazygit/pkg/app.Run({0x12e2020, 0x1ccfb81ea210}, 0x1ccfb839cd00, {{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, ...}})
        /home/hrzlgnm/code/lazygit/pkg/app/app.go:49 +0x145
github.com/jesseduffield/lazygit/pkg/app.Start(0x1ccfb8141f18, {0x0, 0x0})
        /home/hrzlgnm/code/lazygit/pkg/app/entry_point.go:177 +0x1af3
main.main()
        /home/hrzlgnm/code/lazygit/main.go:23 +0xef
```

```log from --log terminal
Aug 14 19:52:23 |INFO| RefsSnapshot took 1.50384ms
Aug 14 19:52:25 |DEBU| RunCommand command="git for-each-ref \"--format=%(objectname) %(refname)\" refs/heads"
Aug 14 19:52:25 |INFO| git for-each-ref "--format=%(objectname) %(refname)" refs/heads (wall 1.17746ms, cpu 1.059ms)
Aug 14 19:52:25 |INFO| RefsSnapshot took 1.34519ms
Aug 14 19:52:27 |DEBU| RunCommand command="git for-each-ref \"--format=%(objectname) %(refname)\" refs/heads"
Aug 14 19:52:27 |INFO| git for-each-ref "--format=%(objectname) %(refname)" refs/heads (wall 1.17313ms, cpu 1.061ms)
Aug 14 19:52:27 |INFO| RefsSnapshot took 1.35069ms
Aug 14 19:52:29 |DEBU| RunCommand command="git for-each-ref \"--format=%(objectname) %(refname)\" refs/heads"
Aug 14 19:52:29 |INFO| git for-each-ref "--format=%(objectname) %(refname)" refs/heads (wall 1.046091ms, cpu 937µs)
Aug 14 19:52:29 |INFO| RefsSnapshot took 1.283011ms
Aug 14 19:52:48 |DEBU| RunCommand command="git --version"
Aug 14 19:52:48 |INFO| git --version (860.52µs)
Aug 14 19:52:48 |DEBU| RunCommand command="git -C /home/hrzlgnm/code/test rev-parse --path-format=absolute --show-toplevel --absolute-git-dir --git-common-dir --show-superproject-working-tree"
Aug 14 19:52:48 |INFO| git -C /home/hrzlgnm/code/test rev-parse --path-format=absolute --show-toplevel --absolute-git-dir --git-common-dir --show-superproject-working-tree (wall 1.715771ms, cpu 1.655ms)
Aug 14 19:52:48 |DEBU| RunCommand command="git -C /home/hrzlgnm/code/test rev-parse --path-format=absolute --show-toplevel --absolute-git-dir --git-common-dir --show-superproject-working-tree"
Aug 14 19:52:48 |INFO| git -C /home/hrzlgnm/code/test rev-parse --path-format=absolute --show-toplevel --absolute-git-dir --git-common-dir --show-superproject-working-tree (wall 1.476461ms, cpu 1.424ms)
Aug 14 19:52:48 |INFO| User-event queue reached a new high-water mark: 1
Aug 14 19:52:48 |INFO| starting main loop
Aug 14 19:52:48 |INFO| Binary is not built with the buildBinary flag so we won't check for an update
Aug 14 19:52:48 |INFO| User-event queue reached a new high-water mark: 2
Aug 14 19:52:48 |INFO| User-event queue reached a new high-water mark: 3
Aug 14 19:52:48 |INFO| refreshing all scopes (async)
Aug 14 19:52:48 |DEBU| RunCommand command="git for-each-ref \"--format=%(objectname) %(refname)\" refs/heads"
Aug 14 19:52:48 |INFO| git for-each-ref "--format=%(objectname) %(refname)" refs/heads (wall 740.77µs, cpu 682µs)
Aug 14 19:52:48 |INFO| RefsSnapshot took 806.11µs
Aug 14 19:52:48 |INFO| User-event queue reached a new high-water mark: 4
Aug 14 19:52:48 |DEBU| RunCommand command="git for-each-ref --sort=-committerdate --format=%(HEAD)%00%(refname:short)%00%(upstream:short)%00%(upstream:track)%00%(push:track)%00%(subject)%00%(objectname)%00%(committerdate:unix) refs/heads"
Aug 14 19:52:48 |DEBU| RunCommand command="git branch --show-current"
Aug 14 19:52:48 |DEBU| RunCommand command="git config --local --get-regexp ^remote\\.[^.]+\\.(url|pushurl)$"
Aug 14 19:52:48 |DEBU| RunCommand command="git stash list -z --pretty=%H|%ct|%gs"
Aug 14 19:52:48 |INFO| User-event queue reached a new high-water mark: 5
Aug 14 19:52:48 |INFO| refreshed merge conflicts in 326.41µs
Aug 14 19:52:48 |DEBU| RunCommand command="git tag --list -n --sort=-creatordate"
Aug 14 19:52:48 |INFO| git for-each-ref --sort=-committerdate --format=%(HEAD)%00%(refname:short)%00%(upstream:short)%00%(upstream:track)%00%(push:track)%00%(subject)%00%(objectname)%00%(committerdate:unix) refs/heads (wall 1.27258ms, cpu 990µs)
Aug 14 19:52:48 |INFO| git for-each-ref --sort=-committerdate --format=%(refname) refs/remotes (1.098921ms)
Aug 14 19:52:48 |INFO| git branch --show-current (wall 1.590961ms, cpu 823µs)
Aug 14 19:52:48 |INFO| git -c log.showSignature=false log -g --format=+%H%x00%ct%x00%gs%x00%P (1.394121ms)
Aug 14 19:52:48 |INFO| User-event queue reached a new high-water mark: 6
Aug 14 19:52:48 |INFO| User-event queue reached a new high-water mark: 7
Aug 14 19:52:48 |INFO| refreshed reflog in 1.823451ms
Aug 14 19:52:48 |DEBU| RunCommand command="git rev-parse --symbolic-full-name main@{u}"
Aug 14 19:52:48 |DEBU| RunCommand command="git config --local --get-regexp ^branch\\."
Aug 14 19:52:48 |DEBU| Error getting git config value for key: status.showUntrackedFiles. Error: the key is not found for [git config --get --null status.showUntrackedFiles]
Aug 14 19:52:48 |INFO| git stash list -z --pretty=%H|%ct|%gs (wall 1.465071ms, cpu 805µs)
Aug 14 19:52:48 |DEBU| RunCommand command="git status --untracked-files=all --porcelain -z --find-renames=50%"
Aug 14 19:52:48 |INFO| User-event queue reached a new high-water mark: 8
Aug 14 19:52:48 |INFO| User-event queue reached a new high-water mark: 9
Aug 14 19:52:48 |INFO| refreshed stash in 2.199091ms
Aug 14 19:52:48 |DEBU| RunCommand command="git rev-parse --symbolic-full-name master@{u}"
Aug 14 19:52:48 |ERRO| <no msg> command="git config --local --get-regexp ^remote\\.[^.]+\\.(url|pushurl)$"
Aug 14 19:52:48 |INFO| git config --local --get-regexp ^remote\.[^.]+\.(url|pushurl)$ (wall 2.155621ms, cpu 808µs)
Aug 14 19:52:48 |INFO| User-event queue reached a new high-water mark: 10
Aug 14 19:52:48 |INFO| User-event queue reached a new high-water mark: 11
Aug 14 19:52:48 |INFO| User-event queue reached a new high-water mark: 12
Aug 14 19:52:48 |INFO| refreshed remotes in 2.413541ms
Aug 14 19:52:48 |INFO| git log HEAD --topo-order --oneline --pretty=format:+%H%x00%at%x00%aN%x00%ae%x00%P%x00%m%x00%D%x00%s --abbrev=40 -300 --no-show-signature -- (1.21966ms)
Aug 14 19:52:48 |INFO| git tag --list -n --sort=-creatordate (wall 2.682681ms, cpu 885µs)
Aug 14 19:52:48 |INFO| User-event queue reached a new high-water mark: 13
Aug 14 19:52:48 |INFO| User-event queue reached a new high-water mark: 14
Aug 14 19:52:48 |INFO| refreshed tags in 3.319521ms
Aug 14 19:52:48 |ERRO| fatal: no upstream configured for branch 'main'
 command="git rev-parse --symbolic-full-name main@{u}"
Aug 14 19:52:48 |INFO| git rev-parse --symbolic-full-name main@{u} (wall 1.4627ms, cpu 833µs)
Aug 14 19:52:48 |DEBU| RunCommand command="git rev-parse --verify --quiet refs/remotes/origin/main"
Aug 14 19:52:48 |ERRO| <no msg> command="git config --local --get-regexp ^branch\\."
Aug 14 19:52:48 |INFO| git config --local --get-regexp ^branch\. (wall 1.59791ms, cpu 761µs)
Aug 14 19:52:48 |DEBU| RunCommand command="git worktree list --porcelain"
Aug 14 19:52:48 |ERRO| fatal: no such branch: 'master'
 command="git rev-parse --symbolic-full-name master@{u}"
Aug 14 19:52:48 |INFO| git rev-parse --symbolic-full-name master@{u} (wall 1.59061ms, cpu 847µs)
Aug 14 19:52:48 |DEBU| RunCommand command="git rev-parse --verify --quiet refs/remotes/origin/master"
Aug 14 19:52:48 |ERRO| <no msg> command="git rev-parse --verify --quiet refs/remotes/origin/main"
Aug 14 19:52:48 |INFO| git rev-parse --verify --quiet refs/remotes/origin/main (wall 1.022261ms, cpu 815µs)
Aug 14 19:52:48 |DEBU| RunCommand command="git rev-parse --verify --quiet refs/heads/main"
Aug 14 19:52:48 |INFO| git worktree list --porcelain (wall 954.731µs, cpu 808µs)
Aug 14 19:52:48 |ERRO| <no msg> command="git rev-parse --verify --quiet refs/remotes/origin/master"
Aug 14 19:52:48 |INFO| git rev-parse --verify --quiet refs/remotes/origin/master (wall 1.048671ms, cpu 779µs)
Aug 14 19:52:48 |DEBU| RunCommand command="git rev-parse --verify --quiet refs/heads/master"
Aug 14 19:52:48 |DEBU| RunCommand command="git -C /home/hrzlgnm/code/test rev-parse --path-format=absolute --absolute-git-dir"
Aug 14 19:52:48 |INFO| git rev-parse --verify --quiet refs/heads/main (wall 867.98µs, cpu 771µs)
Aug 14 19:52:48 |ERRO| <no msg> command="git rev-parse --verify --quiet refs/heads/master"
Aug 14 19:52:48 |INFO| git rev-parse --verify --quiet refs/heads/master (wall 835.26µs, cpu 747µs)
Aug 14 19:52:48 |DEBU| RunCommand command="git rev-list refs/heads/main ^main@{u} ^refs/heads/main"
Aug 14 19:52:48 |DEBU| RunCommand command="git rev-list HEAD ^refs/heads/main"
Aug 14 19:52:48 |INFO| git -C /home/hrzlgnm/code/test rev-parse --path-format=absolute --absolute-git-dir (wall 960.99µs, cpu 811µs)
Aug 14 19:52:48 |INFO| User-event queue reached a new high-water mark: 15
Aug 14 19:52:48 |INFO| User-event queue reached a new high-water mark: 16
Aug 14 19:52:48 |INFO| User-event queue reached a new high-water mark: 17
Aug 14 19:52:48 |INFO| refreshed branches in 6.182312ms
Aug 14 19:52:48 |INFO| User-event queue reached a new high-water mark: 18
Aug 14 19:52:48 |INFO| refreshed pull requests in 24.04µs
Aug 14 19:52:48 |INFO| git rev-list HEAD ^refs/heads/main (1.03424ms)
Aug 14 19:52:48 |INFO| git rev-list refs/heads/main ^main@{u} ^refs/heads/main (1.27495ms)
Aug 14 19:52:48 |ERRO| fatal: no upstream configured for branch 'main'
 command="git rev-list refs/heads/main ^main@{u} ^refs/heads/main"
Aug 14 19:52:48 |INFO| User-event queue reached a new high-water mark: 19
Aug 14 19:52:48 |INFO| User-event queue reached a new high-water mark: 20
Aug 14 19:52:48 |INFO| refreshed commits and commit files in 7.441353ms
Aug 14 19:52:48 |INFO| Receiving focus - refreshing
Aug 14 19:52:48 |INFO| refreshing all scopes (async)
Aug 14 19:52:48 |DEBU| RunCommand command="git for-each-ref \"--format=%(objectname) %(refname)\" refs/heads"
Aug 14 19:52:48 |INFO| git for-each-ref "--format=%(objectname) %(refname)" refs/heads (wall 1.0392ms, cpu 913µs)
Aug 14 19:52:48 |INFO| RefsSnapshot took 1.18208ms
Aug 14 19:52:48 |INFO| postRefreshUpdate for reflogCommits took 112.13µs
Aug 14 19:52:48 |INFO| postRefreshUpdate for stash took 3.97µs
Aug 14 19:52:48 |INFO| postRefreshUpdate for remotes took 3.46µs
Aug 14 19:52:48 |INFO| postRefreshUpdate for remoteBranches took 3.36µs
Aug 14 19:52:48 |INFO| postRefreshUpdate for tags took 3.56µs
Aug 14 19:52:48 |DEBU| RunCommand command="git tag --list -n --sort=-creatordate"
Aug 14 19:52:48 |DEBU| RunCommand command="git config --local --get-regexp ^remote\\.[^.]+\\.(url|pushurl)$"
Aug 14 19:52:48 |DEBU| RunCommand command="git branch --show-current"
Aug 14 19:52:48 |INFO| refreshed merge conflicts in 8.02µs
Aug 14 19:52:48 |DEBU| RunCommand command="git for-each-ref --sort=-committerdate --format=%(HEAD)%00%(refname:short)%00%(upstream:short)%00%(upstream:track)%00%(push:track)%00%(subject)%00%(objectname)%00%(committerdate:unix) refs/heads"
Aug 14 19:52:48 |DEBU| RunCommand command="git fetch --all --no-write-fetch-head"
Aug 14 19:52:48 |INFO| git -c log.showSignature=false log -g --format=+%H%x00%ct%x00%gs%x00%P (2.833041ms)
Aug 14 19:52:48 |INFO| refreshed reflog in 2.890311ms
Aug 14 19:52:48 |ERRO| <no msg> command="git config --local --get-regexp ^remote\\.[^.]+\\.(url|pushurl)$"
Aug 14 19:52:48 |INFO| git config --local --get-regexp ^remote\.[^.]+\.(url|pushurl)$ (wall 2.4198ms, cpu 814µs)
Aug 14 19:52:48 |DEBU| RunCommand command="git stash list -z --pretty=%H|%ct|%gs"
Aug 14 19:52:48 |INFO| git tag --list -n --sort=-creatordate (wall 9.094553ms, cpu 857µs)
Aug 14 19:52:48 |INFO| refreshed tags in 9.250553ms
Aug 14 19:52:48 |INFO| git branch --show-current (wall 7.614572ms, cpu 829µs)
Aug 14 19:52:48 |INFO| git for-each-ref --sort=-committerdate --format=%(refname) refs/remotes (6.144022ms)
Aug 14 19:52:48 |INFO| refreshed remotes in 8.208372ms
Aug 14 19:52:48 |INFO| git for-each-ref --sort=-committerdate --format=%(HEAD)%00%(refname:short)%00%(upstream:short)%00%(upstream:track)%00%(push:track)%00%(subject)%00%(objectname)%00%(committerdate:unix) refs/heads (wall 7.338852ms, cpu 1.005ms)
Aug 14 19:52:48 |DEBU| RunCommand command="git rev-list HEAD ^refs/heads/main"
Aug 14 19:52:48 |INFO| git stash list -z --pretty=%H|%ct|%gs (wall 5.780042ms, cpu 799µs)
Aug 14 19:52:48 |INFO| refreshed stash in 5.860182ms
Aug 14 19:52:48 |INFO| postRefreshUpdate for localBranches took 110.141µs
Aug 14 19:52:48 |ERRO| read /dev/ptmx: input/output error
Aug 14 19:52:48 |DEBU| RunCommand command="git config --local --get-regexp ^branch\\."
Aug 14 19:52:48 |DEBU| RunCommand command="git rev-list refs/heads/main ^main@{u} ^refs/heads/main"
Aug 14 19:52:48 |INFO| git fetch --all --no-write-fetch-head (7.530642ms)
Aug 14 19:52:48 |INFO| refreshing the following scopes (sync): branches,commits,remotes,tags,pullRequests,worktrees
Aug 14 19:52:48 |INFO| git rev-list HEAD ^refs/heads/main (3.236812ms)
Aug 14 19:52:48 |ERRO| <no msg> command="git config --local --get-regexp ^branch\\."
Aug 14 19:52:48 |INFO| git config --local --get-regexp ^branch\. (wall 3.294481ms, cpu 835µs)
Aug 14 19:52:48 |DEBU| RunCommand command="git worktree list --porcelain"
Aug 14 19:52:48 |DEBU| Error getting git config value for key: status.showUntrackedFiles. Error: the key is not found for [git config --get --null status.showUntrackedFiles]
Aug 14 19:52:48 |DEBU| RunCommand command="git status --untracked-files=all --porcelain -z --find-renames=50%"
Aug 14 19:52:48 |INFO| git worktree list --porcelain (wall 1.27918ms, cpu 799µs)
Aug 14 19:52:48 |DEBU| RunCommand command="git -C /home/hrzlgnm/code/test rev-parse --path-format=absolute --absolute-git-dir"
Aug 14 19:52:48 |DEBU| using cache for key rebase.updateRefs
Aug 14 19:52:48 |INFO| git log HEAD --topo-order --oneline --pretty=format:+%H%x00%at%x00%aN%x00%ae%x00%P%x00%m%x00%D%x00%s --abbrev=40 -300 --no-show-signature -- (3.681981ms)
Aug 14 19:52:48 |INFO| git rev-list refs/heads/main ^main@{u} ^refs/heads/main (3.351161ms)
Aug 14 19:52:48 |ERRO| fatal: no upstream configured for branch 'main'
 command="git rev-list refs/heads/main ^main@{u} ^refs/heads/main"
Aug 14 19:52:48 |INFO| refreshed commits and commit files in 13.728994ms
Aug 14 19:52:48 |INFO| postRefreshUpdate for commits took 8.756233ms
Aug 14 19:52:48 |INFO| postRefreshUpdate for worktrees took 53.89µs
Aug 14 19:52:48 |INFO| postRefreshUpdate for reflogCommits took 48.1µs
Aug 14 19:52:48 |INFO| postRefreshUpdate for tags took 3.81µs
Aug 14 19:52:48 |INFO| postRefreshUpdate for remotes took 3.75µs
Aug 14 19:52:48 |INFO| postRefreshUpdate for remoteBranches took 3.03µs
Aug 14 19:52:48 |INFO| postRefreshUpdate for stash took 3.46µs
Aug 14 19:52:48 |DEBU| using cache for key rebase.updateRefs
Aug 14 19:52:48 |INFO| git -C /home/hrzlgnm/code/test rev-parse --path-format=absolute --absolute-git-dir (wall 3.901562ms, cpu 853µs)
Aug 14 19:52:48 |INFO| postRefreshUpdate for commits took 118.97µs
Aug 14 19:52:48 |INFO| refreshed pull requests in 11.62µs
Aug 14 19:52:48 |INFO| refreshed branches in 16.949066ms
Aug 14 19:52:48 |DEBU| using cache for key rebase.updateRefs
Aug 14 19:52:48 |DEBU| RunCommand command="git for-each-ref \"--format=%(objectname) %(refname)\" refs/heads"
Aug 14 19:52:48 |INFO| postRefreshUpdate for localBranches took 29.33µs
Aug 14 19:52:48 |INFO| postRefreshUpdate for worktrees took 24.18µs
Aug 14 19:52:48 |DEBU| using cache for key rebase.updateRefs
Aug 14 19:52:48 |DEBU| using cache for key rebase.updateRefs
Aug 14 19:52:48 |INFO| git for-each-ref "--format=%(objectname) %(refname)" refs/heads (wall 6.056072ms, cpu 1.111ms)
Aug 14 19:52:48 |INFO| RefsSnapshot took 6.285632ms
Aug 14 19:52:48 |DEBU| RunCommand command="git tag --list -n --sort=-creatordate"
Aug 14 19:52:48 |DEBU| RunCommand command="git branch --show-current"
Aug 14 19:52:48 |DEBU| RunCommand command="git for-each-ref --sort=-committerdate --format=%(HEAD)%00%(refname:short)%00%(upstream:short)%00%(upstream:track)%00%(push:track)%00%(subject)%00%(objectname)%00%(committerdate:unix) refs/heads"
Aug 14 19:52:48 |INFO| git tag --list -n --sort=-creatordate (wall 5.518092ms, cpu 1.073ms)
Aug 14 19:52:48 |INFO| refreshed tags in 5.719262ms
Aug 14 19:52:48 |INFO| postRefreshUpdate for tags took 11.81µs
Aug 14 19:52:48 |DEBU| RunCommand command="git config --local --get-regexp ^remote\\.[^.]+\\.(url|pushurl)$"
Aug 14 19:52:48 |INFO| git branch --show-current (wall 8.010833ms, cpu 921µs)
Aug 14 19:52:48 |INFO| git -c log.showSignature=false log -g --format=+%H%x00%ct%x00%gs%x00%P (6.605632ms)
Aug 14 19:52:48 |INFO| git for-each-ref --sort=-committerdate --format=%(HEAD)%00%(refname:short)%00%(upstream:short)%00%(upstream:track)%00%(push:track)%00%(subject)%00%(objectname)%00%(committerdate:unix) refs/heads (wall 6.863292ms, cpu 1.239ms)
Aug 14 19:52:48 |DEBU| RunCommand command="git rev-list refs/heads/main ^main@{u} ^refs/heads/main"
Aug 14 19:52:48 |DEBU| RunCommand command="git config --local --get-regexp ^branch\\."
Aug 14 19:52:48 |INFO| refreshed reflog in 6.698452ms
Aug 14 19:52:48 |INFO| git for-each-ref --sort=-committerdate --format=%(refname) refs/remotes (4.793982ms)
Aug 14 19:52:48 |ERRO| <no msg> command="git config --local --get-regexp ^remote\\.[^.]+\\.(url|pushurl)$"
Aug 14 19:52:48 |INFO| git config --local --get-regexp ^remote\.[^.]+\.(url|pushurl)$ (wall 7.528633ms, cpu 987µs)
Aug 14 19:52:48 |DEBU| RunCommand command="git rev-list HEAD ^refs/heads/main"
Aug 14 19:52:48 |INFO| postRefreshUpdate for reflogCommits took 106.26µs
Aug 14 19:52:48 |INFO| refreshed remotes in 7.807113ms
Aug 14 19:52:48 |INFO| git rev-list refs/heads/main ^main@{u} ^refs/heads/main (2.571091ms)
Aug 14 19:52:48 |ERRO| fatal: no upstream configured for branch 'main'
 command="git rev-list refs/heads/main ^main@{u} ^refs/heads/main"
Aug 14 19:52:48 |INFO| git log HEAD --topo-order --oneline --pretty=format:+%H%x00%at%x00%aN%x00%ae%x00%P%x00%m%x00%D%x00%s --abbrev=40 -300 --no-show-signature -- (2.606981ms)
Aug 14 19:52:48 |INFO| git rev-list HEAD ^refs/heads/main (1.37254ms)
Aug 14 19:52:48 |INFO| refreshed commits and commit files in 14.643635ms
Aug 14 19:52:48 |INFO| postRefreshUpdate for remotes took 7.49µs
Aug 14 19:52:48 |INFO| postRefreshUpdate for remoteBranches took 3.54µs
Aug 14 19:52:48 |DEBU| using cache for key rebase.updateRefs
Aug 14 19:52:48 |INFO| postRefreshUpdate for commits took 81.02µs
Aug 14 19:52:48 |ERRO| <no msg> command="git config --local --get-regexp ^branch\\."
Aug 14 19:52:48 |INFO| git config --local --get-regexp ^branch\. (wall 4.064781ms, cpu 878µs)
Aug 14 19:52:48 |DEBU| RunCommand command="git worktree list --porcelain"
Aug 14 19:52:48 |DEBU| using cache for key rebase.updateRefs
Aug 14 19:52:48 |INFO| git worktree list --porcelain (wall 1.090081ms, cpu 822µs)
Aug 14 19:52:48 |DEBU| RunCommand command="git -C /home/hrzlgnm/code/test rev-parse --path-format=absolute --absolute-git-dir"
Aug 14 19:52:48 |INFO| git -C /home/hrzlgnm/code/test rev-parse --path-format=absolute --absolute-git-dir (wall 1.50025ms, cpu 757µs)
Aug 14 19:52:48 |INFO| refreshed branches in 18.580916ms
Aug 14 19:52:48 |INFO| Refresh took 34.865442ms
Aug 14 19:52:48 |INFO| refreshed pull requests in 4.49µs
Aug 14 19:52:48 |DEBU| using cache for key rebase.updateRefs
Aug 14 19:52:48 |INFO| postRefreshUpdate for localBranches took 20.85µs
Aug 14 19:52:48 |INFO| postRefreshUpdate for worktrees took 26.71µs
````

--- a/pkg/gui/controllers/files_controller.go
+++ b/pkg/gui/controllers/files_controller.go
@@ -130,10 +130,11 @@ func (self *FilesController) GetKeybindings(opts types.KeybindingsOpts) []*types
 			OpensMenu:   true,
 		},
 		{
-			Keys:        opts.GetKeys(opts.Config.Files.ToggleStagedAll),
-			Handler:     self.toggleStagedAll,
-			Description: self.c.Tr.ToggleStagedAll,
-			Tooltip:     self.c.Tr.ToggleStagedAllTooltip,
+			Keys:              opts.GetKeys(opts.Config.Files.ToggleStagedAll),
+			Handler:           self.toggleStagedAll,
+			GetDisabledReason: self.require(self.anyFilesDisplayed),
+			Description:       self.c.Tr.ToggleStagedAll,
+			Tooltip:           self.c.Tr.ToggleStagedAllTooltip,
 		},
 		{
 			Keys:              opts.GetKeys(opts.Config.Universal.GoInto),
@@ -914,6 +915,17 @@ func (self *FilesController) openSubmoduleConflictMenu(file *models.File) error 
 			},
 		},
 	})
+}
+
+// The stage-all command acts on the file tree as it is displayed, so there has
+// to be something in it. This is also the case before the first files refresh
+// has come in, when there is no tree at all yet.
+func (self *FilesController) anyFilesDisplayed() *types.DisabledReason {
+	if self.context().FileTreeViewModel.Len() == 0 {
+		return &types.DisabledReason{Text: self.c.Tr.NoChangedFiles}
+	}
+
+	return nil
 }
 
 func (self *FilesController) toggleStagedAll() error {

--- a/pkg/integration/tests/file/stage_all_without_changed_files.go
+++ b/pkg/integration/tests/file/stage_all_without_changed_files.go
@@ -1,0 +1,28 @@
+package file
+
+import (
+	"github.com/jesseduffield/lazygit/pkg/config"
+	. "github.com/jesseduffield/lazygit/pkg/integration/components"
+)
+
+var StageAllWithoutChangedFiles = NewIntegrationTest(NewIntegrationTestArgs{
+	Description:  "Pressing the stage-all key when there are no changed files says that there are none",
+	ExtraCmdArgs: []string{},
+	Skip:         false,
+	SetupConfig:  func(config *config.AppConfig) {},
+	SetupRepo: func(shell *Shell) {
+		shell.EmptyCommit("first commit")
+	},
+	Run: func(t *TestDriver, keys config.KeybindingConfig) {
+		t.Views().Files().
+			IsFocused().
+			IsEmpty().
+			Press(keys.Files.ToggleStagedAll).
+			Tap(func() {
+				/* EXPECTED:
+				t.ExpectToast(Contains("No changed files"))
+				ACTUAL: */
+				t.ExpectPopup().Alert().Title(Equals("Error")).Content(Contains("Nothing to stage")).Confirm()
+			})
+	},
+})

--- a/pkg/integration/tests/file/stage_all_without_changed_files.go
+++ b/pkg/integration/tests/file/stage_all_without_changed_files.go
@@ -19,10 +19,7 @@ var StageAllWithoutChangedFiles = NewIntegrationTest(NewIntegrationTestArgs{
 			IsEmpty().
 			Press(keys.Files.ToggleStagedAll).
 			Tap(func() {
-				/* EXPECTED:
 				t.ExpectToast(Contains("No changed files"))
-				ACTUAL: */
-				t.ExpectPopup().Alert().Title(Equals("Error")).Content(Contains("Nothing to stage")).Confirm()
 			})
 	},
 })

--- a/pkg/integration/tests/test_list.go
+++ b/pkg/integration/tests/test_list.go
@@ -248,6 +248,7 @@ var tests = []*components.IntegrationTest{
 	file.RenameSimilarityThresholdChange,
 	file.RenamedFiles,
 	file.RenamedFilesNoRootItem,
+	file.StageAllWithoutChangedFiles,
 	file.StageChildrenRangeSelect,
 	file.StageDeletedRangeSelect,
 	file.StageRangeSelect,

